scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py

```diff
@@ -109,10 +109,20 @@ def __init__(self, parent=None):
         self.electron_peaks = self._get_electron_peaks()
         self._generate_element_widgets()
 
-    # Return an unused colour, if all used then start with the first one
-    def get_color(self):
-        color = "C%d" % self.color_index
-        self.color_index = (self.color_index + 1) % self.num_colors
+    # Return the first unused colour, if all used then restart from the beginning of the cycle
+    def get_color(self, element):
+        if element in self.used_colors:
+            return self.used_colors[element]
+
+        occurrences = []
+        for i in range(self.num_colors):
+            occurrences.append(self.used_colors.values().count('C{}'.format(i)))
+
+        color_list = ['C{}'.format(i) for i in range(self.num_colors) if occurrences[i] == min(occurrences)]
+        color_index = min([int(col[1:]) for col in color_list])
```

Comment:
```suggestion
        color_index = occurrences.index(min(occurrences))
```
I think this does the same, the `color_list` line can then be deleted